llama-index-core/llama_index/core/workflow/workflow.py

```diff
@@ -48,6 +50,8 @@ def __init__(
         # Context management
         self._root_context: Context = Context()
         self._step_to_context: Dict[str, Context] = {}
+        # Services management
+        self._service_manager = service_manager
 
     @classmethod
     def add_step(cls, func: Callable) -> None:
```

Comment:
if we `add_step` that introduces a new `Workflow`/`service`, would we have to register it with `_service_manager`?

Reply:
Yea, ideally you've already launched it before introducing a step that depends on it (I think you'd also have to call `add_service()` after adding a free-function that uses a service)

Reply:
makes sense!